src/consensus/consensus.rs

```diff
@@ -22,6 +21,13 @@ pub enum SystemMessage {
     DecidedValueForReadNode(proto::DecidedValue),
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ValidatorSetConfig {
+    pub effective_at: u64,
+    pub validator_public_keys: Vec<String>,
+    pub shard_ids: Vec<u32>,
```

Comment:
What is the goal with configuring these? Should the validator set only be deemed fit for the shard indices in the array?

Reply:
Is it meant to be a replacement for the existing `shard_ids` configuration?

Reply:
Yes, it should only be valid for the shard ids provided. Right now we have the same validator sets for all shards, but it should be possible to have completely separate validator sets for each shard. And change them over time (at specific heights)